### PR TITLE
Extend the list of supported pip versions

### PR DESCRIPTION
**Pull request by frenzymadness**
This change extends the list of supported versions of pip in our CI. I know it does not make sense to include all available releases of pip there so I included only the ones I consider important from the (RHEL) maintenance point of view and added comments for our future selfs.

Version 9 is important because it's the default in RHEL 8 (platform-python-pip) and micropipenv will need to work flawlessly with this version when we package it to RHEL.
We updated pip from 19.2.3 to 19.3.1 after we released Python 3.8 module so it would be nice to support both versions for some time.

By the way, micropipenv has `install_requires=["pip>=9"]` in its setup.py so any incompatibility uncovered by this change should be fixed rather sooner than later. If we decide to not support some specific version of pip for some reason, we should exclude that version in `install_requires`.

## This introduces a breaking change

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by fridex** [comment]
Very nice! :+1: 

> By the way, micropipenv has `install_requires=["pip>=9"]` in its setup.py so any incompatibility uncovered by this change should be fixed rather sooner than later. If we decide to not support some specific version of pip for some reason, we should exclude that version in `install_requires`.

Honestly, I did not check all the versions. I tried older fedora containers (I think down to f27 based on my sh history but not sure if the last worked well though). ubi8 worked well for me, ubi7 was too old IIRC (hence `pip>=9`).

**Comment by fridex** [comment]
Briefly looking at the log, pip-19.2 had some changes in the editable API. It should be fairly doable to add support.

For pip-9.0, case the `ValueError: invalid truth value '>=9.0,<10.0'` error sounds strange at first look.

**Comment by fridex** [comment]
> For pip-9.0, case the `ValueError: invalid truth value '>=9.0,<10.0'` error sounds strange at first look.

BTW it might be due to `PIP_VERSION` environment variable collision. Are you ok with renaming it to something like `MICROPIPENV_TEST_PIP_VERSION`?

**Comment by frenzymadness** [comment]
> > For pip-9.0, case the `ValueError: invalid truth value '>=9.0,<10.0'` error sounds strange at first look.
> 
> BTW it might be due to `PIP_VERSION` environment variable collision. Are you ok with renaming it to something like `MICROPIPENV_TEST_PIP_VERSION`?

You are right, thank you.

```
$ pip --version
pip 9.0.3 from /home/lbalhar/.virtualenvs/tmp-cd8b4dae64c0d30/lib/python3.6/site-packages (python 3.6)
$ pip search tldr
tldr (1.0.0)         - command line client for tldr
snippy-tldr (0.2.0)  - Snippy plugin to import tldr man pages.
pytest-tldr (0.2.1)  - A pytest plugin that limits the output to just the things you need.
tldr.py (0.8.0)      - A python client for tldr: simplified and community-driven man pages.
You are using pip version 9.0.3, however version 20.1.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
$ PIP_VERSION=19.3.1 pip search tldr
Traceback (most recent call last):
  File "/home/lbalhar/.virtualenvs/tmp-cd8b4dae64c0d30/bin/pip", line 8, in <module>
    sys.exit(main())
  File "/home/lbalhar/.virtualenvs/tmp-cd8b4dae64c0d30/lib/python3.6/site-packages/pip/__init__.py", line 234, in main
    cmd_name, cmd_args = parseopts(args)
  File "/home/lbalhar/.virtualenvs/tmp-cd8b4dae64c0d30/lib/python3.6/site-packages/pip/__init__.py", line 183, in parseopts
    general_options, args_else = parser.parse_args(args)
  File "/usr/lib64/python3.6/optparse.py", line 1371, in parse_args
    values = self.get_default_values()
  File "/home/lbalhar/.virtualenvs/tmp-cd8b4dae64c0d30/lib/python3.6/site-packages/pip/baseparser.py", line 283, in get_default_values
    defaults = self._update_defaults(self.defaults.copy())  # ours
  File "/home/lbalhar/.virtualenvs/tmp-cd8b4dae64c0d30/lib/python3.6/site-packages/pip/baseparser.py", line 230, in _update_defaults
    val = strtobool(val)
  File "/usr/lib64/python3.6/distutils/util.py", line 317, in strtobool
    raise ValueError("invalid truth value %r" % (val,))
ValueError: invalid truth value '19.3.1'
```

I'll try to rename the variable.

**Comment by frenzymadness** [comment]
The last commit fixed the compatibility with pip 9.

**Comment by fridex** [comment]
> The last commit fixed the compatibility with pip 9.

Nice! :clap: 

What about commenting out pip-19.2 tests for now? I'll try to look at the support in a separate PR if you are fine with that.

**Comment by frenzymadness** [comment]
> > The last commit fixed the compatibility with pip 9.
> 
> Nice! clap
> 
> What about commenting out pip-19.2 tests for now? I'll try to look at the support in a separate PR if you are fine with that.

How long do you think it will take to fix it? I am not against disabling it now but a fix would help me with the CI in the python container.

**Comment by fridex** [comment]
> > > The last commit fixed the compatibility with pip 9.
> > 
> > 
> > Nice! clap
> > What about commenting out pip-19.2 tests for now? I'll try to look at the support in a separate PR if you are fine with that.
> 
> How long do you think it will take to fix it? I am not against disabling it now but a fix would help me with the CI in the python container.

OK, I can have a look at it later today. :+1:

**Comment by fridex** [comment]
> > > > The last commit fixed the compatibility with pip 9.
> > > 
> > > 
> > > Nice! clap
> > > What about commenting out pip-19.2 tests for now? I'll try to look at the support in a separate PR if you are fine with that.
> > 
> > 
> > How long do you think it will take to fix it? I am not against disabling it now but a fix would help me with the CI in the python container.
> 
> OK, I can have a look at it later today.

Done in #92.

**Comment by fridex** [comment]
... it looks like CI is giving this green flag. Let's merge this :+1: Thanks! :clap:
